### ISSUE TITLE
open_all removes dom elements when internal operation fails.

### ISSUE THREAD

**Issue by Spookenator** [vakata/jstree#1412]
Hi Ivan,

I hope you are keeping well. 
I ran into an interesting problem when trying to open_all on a huge branch. It seems that the internal recursive function gets a heap overflow error and returns then returns an empty branch to the tree.

I was wondering if it is possible open all children using a jquery selector rather than trying to traverse the tree?

Let me know what you think.
Regards,
Eduard

**Comment by vakata** [comment]
Hi,

The nodes are not kept in the DOM, so a jQuery selector will likely fail (if I understand the suggestion correctly). As for the issue - if it is AJAX loading related - return a nested structure as a temporary fix. 

Could you please provide a jsfiddle for example, so that I can investigate and fix the issue. What amount of nodes are we talking about?

Best regards,
Ivan

**Comment by Spookenator** [comment]
Hi Ivan

The JsFiddle as requested:
https://jsfiddle.net/jLomru2d/1/

**Comment by vakata** [comment]
Hi, 
Thank you - the fiddle really helped in identifying the problem - it is not with jstree, but with the data you provide. 

I did not encounter a heap overflow, what I did find though is duplicate IDs - for example `8421`. jsTree does not work with duplicate IDs. 

If you remove all IDs from the data you provided (so that jstree will autogenerate IDs) you will see that `open_all` button works normally.

Best regards,
Ivan
